class.jetpack.php

```diff
@@ -1302,8 +1302,8 @@ public static function is_development_mode() {
 			$development_mode = JETPACK_DEV_DEBUG;
 		}
 
-		elseif ( site_url() && false === strpos( site_url(), '.' ) ) {
-			$development_mode = true;
+		elseif ( $site_url = site_url() ) {
```

Comment:
I'm not sure why we even need to check if `site_url()` returns something (something would be pretty broken if it didn't), but it's possible it's this way for a reason.